mirage/start.js

```diff
@@ -0,0 +1,38 @@
+import Mirage from 'mirage-server';
+import baseConfig from './config';
+import camelCase from 'lodash/camelCase';
+import '../tests/force-fetch-polyfill';
+
+const environment = process.env.NODE_ENV;
+const moduleTypes = ['factories', 'fixtures', 'scenarios', 'models', 'serializers', 'identity-managers'];
+
+const req = require.context('./', true, /\.js$/);
+
+let modules = moduleTypes.reduce((memo, name) => {
+  memo[camelCase(name)] = {};
+  return memo;
+}, {});
+
+req.keys().forEach((modulePath) => {
+  const moduleParts = modulePath.split('/');
+  const moduleType = moduleParts[1];
+  const moduleName = moduleParts[2];
+
+  if (moduleName) {
+    const moduleKey = camelCase(moduleName.replace('.js', ''));
+    modules[moduleType][moduleKey] = req(modulePath).default;
+  }
+});
+
+export function startDevMirage() {
+  let options = Object.assign(modules, { environment, baseConfig });
+  return new Mirage(options);
+}
+
+export function startTestMirage() {
+  let testModules = { ...modules };
+  delete modules.scenarios;
```

Comment:
it'd probably be slightly better (albeit a bit uglier) to push the module list in from the outside.  what if we pulled all the code above these methods into a generic `start` that takes a module list and then `startTest...` and `startDev...` pass in what they want.  if you wanted to DRY it you could change `moduleTypes` to be those modules common to both and then pass the default concatenated with whatever extra moduleTypes you want for the given use case